### PR TITLE
feat: add better typescript types for pubsub

### PR DESCRIPTION
**Pull request by grant**
Inits better TypeScript types using Quicktype.

- Removes protoc types (they aren't accurate even with Improbable's plug-in)
- Adds a couple JSON payloads seen from prod
- Generate TS interfaces
- Write a JS method (to<Type>) for JS users
- Adds `tsc` step in CI to run before publishing. Modified from `yargs`' setup.

In the future we can use the quicktype module to gen these TS interfaces.

### CODE REVIEW

**Comment by grant** [comment]
I updated the tsc config to work with GitHub CI.

Modified from the setup seen with https://github.com/yargs/yargs

File: README.md [hunk 1]
```diff
@@ -18,10 +18,6 @@ npm i @google/events
...
 
-Generate libraries:
+Generate libraries from quicktype (currently manual).
 
-```sh
-./gen.sh
-```
-
-Observe the new files in `src/`.
+https://app.quicktype.io/
```
Comment (grayside) [inline README.md:23]: I think a bit more documentation is needed on what to put into quicktype.io and the process to collect that information. Would expect just a few bullets for now.

Comment (grant) [reply README.md]: Added a bit more documentation on the process!

**Comment by grant** [comment]
@grayside Thanks for the comment. PTAL :)

**Pull request merged**